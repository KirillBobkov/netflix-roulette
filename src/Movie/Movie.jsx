import React from 'react';
import './Movie.scss'

export default ({ index, info }) => (
   <li className='movies__item'>
       <p className='movies__poster'>
           <img 
                src={`posters/${index + 1}.png`} 
                width='300'
                height='400' 
                alt={info.name} 
           />
        </p>

        <p className='movies__description'>
            <span className='movies__title'>
                {info.name}
            </span>

            <span className='movies__gengre'>
                {info.gengre}
            </span>

            <span className='movies__year'>
                {info.year}
            </span>
        </p>
    </li>
);
