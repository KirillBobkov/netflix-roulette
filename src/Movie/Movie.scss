.movies {

    &__item {
        margin-bottom: 50px;
        width: 300px;
        margin-right: 30px;

        &:nth-child(3n) {
            margin-right: 0px;
        }
    }

    &__poster {
        margin-bottom: 10px;
    }

    &__title {
        display: block;
        font-size: 14px;
        margin-bottom: 10px;
    }

    &__description {
        position: relative;
    }

    &__gengre {
        display: block;
        font-size: 10px;
        color: #8f8f8f;
    }

    &__year {
        position: absolute;
        top: 0;
        right: 0;
        font-size: 10px;
        padding: 4px 10px;
        border: 1px solid #8f8f8f;
        border-radius: 3px;
        color: #8f8f8f;
    }
}