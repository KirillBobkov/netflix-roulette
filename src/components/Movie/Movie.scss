.movies {
    &__item {
        margin-bottom: 40px;
        width: 260px;
        margin-right: 43px;

        &:nth-child(3n) {
            margin-right: 0px;
        }
    }

    &__poster {
        margin-bottom: 17px;
    }

    &__title {
        display: block;
        font-size: 14px;
        margin-bottom: 10px;    
    }

    &__description {
        position: relative;
    }

    &__gengre {
        display: block;
        font-size: 11px;
        color: #8f8f8f;
    }

    &__year {
        position: absolute;
        top: 0;
        right: 2px;
        font-size: 10px;
        padding: 4px 13px;
        border: 1px solid #8f8f8f;
        border-radius: 3px;
        color: #8f8f8f;
    }
}