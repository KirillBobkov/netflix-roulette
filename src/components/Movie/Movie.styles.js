import styled from "styled-components";

export const MovieItem = styled.li`
    margin-bottom: 40px;
    width: 260px;
    margin-right: 43px;

    &:nth-child(3n) {
        margin-right: 0px;
    }
`;

export const MoviePoster = styled.p`
    width: 260px;
    height: 365px;
    margin-bottom: 17px;
`;

export const MovieTitle = styled.span`
    display: block;
    font-size: 14px;
    margin-bottom: 10px;  
    width: 205px;  
`;

export const MovieDescription = styled.p`
    position: relative;
`;

export const MovieGenre = styled.span`
    display: block;
    font-size: 11px;
    color: #8f8f8f;
`;

export const MovieYear = styled.span`
    position: absolute;
    top: 0;
    right: 2px;
    font-size: 10px;
    padding: 4px 13px;
    border: 1px solid #8f8f8f;
    border-radius: 3px;
    color: #8f8f8f;
`;
