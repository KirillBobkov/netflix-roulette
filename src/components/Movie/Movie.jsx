import React from 'react';
import PropTypes from 'prop-types';
import './Movie.scss';
import { getYear } from '../../utils';

export const Movie = ({ movie }) => (
  <li className='movies__item'>
    <p className='movies__poster'>
      <img 
        src={movie.poster_path} 
        width='260'
        height='365' 
        alt={movie.title}
      />
    </p>

    <p className='movies__description'>
      <span className='movies__title'>
        {movie.title}
      </span>

      <span className='movies__gengre'>
        {movie.genres.length && movie.genres.join(', ')}
      </span>

      <span className='movies__year'>
        {getYear(movie.release_date)}
      </span>
    </p>
  </li>
);

Movie.propTypes = {
  movie: PropTypes.shape({
    poster_path: PropTypes.string,
    title: PropTypes.string,
    genres: PropTypes.array,
    release_date: PropTypes.string
  })
};