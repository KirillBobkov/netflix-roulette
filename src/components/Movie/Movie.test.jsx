import React from 'react';
import { Movie } from './Movie';
import { shallow, mount } from 'enzyme';

describe('Movie component', () => {
    let movie;
    let index;

    beforeAll(() => {
        movie = {
            name: 'Bohemian Rhapsody',
            gengre: 'Action & Adventure',
            year: 2003,
            id: 4689453234888769
        };
        index = 1;     
    });

    it('should be render correctly', () => {
        const component = shallow(<Movie key={movie.id} index={index} info={movie} />);
        expect(component).toMatchSnapshot();
    });

    it('should be render correctly with childs', () => {
        const component = mount(<Movie key={movie.id} index={index} info={movie} />);
        expect(component).toMatchSnapshot();
    });

    it('should render a year of movie', () => {
        const component = mount(<Movie key={movie.id} index={index} info={movie} />);
        const year = component.find('.movies__year');
        expect(year.text()).toEqual('2003');
    });

    it('should render a gengre of movie', () => {
        const component = mount(<Movie key={movie.id} index={index} info={movie} />);
        const year = component.find('.movies__gengre');
        expect(year.text()).toEqual('Action & Adventure');
    });

    it('should render a title of movie', () => {
        const component = mount(<Movie key={movie.id} index={index} info={movie} />);
        const year = component.find('.movies__title');
        expect(year.text()).toEqual('Bohemian Rhapsody');
    });
});