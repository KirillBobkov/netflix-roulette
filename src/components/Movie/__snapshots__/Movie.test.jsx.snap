// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component should be render correctly 1`] = `ReactWrapper {}`;

exports[`Movie component should be render correctly with childs 1`] = `ReactWrapper {}`;
